data/data/bootstrap/files/usr/local/bin/bootkube.sh.template

```diff
@@ -201,6 +201,15 @@ echo "etcd cluster up. Killing etcd certificate signer..."
 
 podman rm --force etcd-signer
 rm --force /etc/kubernetes/manifests/machineconfigoperator-bootstrap-pod.yaml
+# Copy the bootstrap MCs to inject into the target cluster
+# Yes this is a brutal hack, need to improve the MCC bootstrap above
+# 9a so we're after 99 - should change the others to 50- or something?
+for x in /etc/mcs/bootstrap/machine-configs/*.yaml; do
```

Comment:
Where do these come from in the first place?  Looks like `/etc/mcs` has only been mentioned in 6da1973fb (#879), but I don't see any installer code obviously populating it.  Maybe it's [here][1]?  If so, is there a reason you couldn't address this entirely within openshift/machine-config-operator?

[1]: https://github.com/openshift/machine-config-operator/pull/226/files#diff-51bc64f09fd6698cbb13c264c6b29e24R13

Reply:
My understanding of the current code is here https://github.com/openshift/machine-config-operator/issues/367#issuecomment-460368011

> but I don't see any installer code obviously populating it. 

AIUI it's the static pod which we are just removing above this code:
`/etc/kubernetes/manifests/machineconfigoperator-bootstrap-pod.yaml`

> If so, is there a reason you couldn't address this entirely within openshift/machine-config-operator?

Maybe - I didn't write this code and am still learning things here, so alternative suggestions appreciated!

Reply:
As far as I understand things though...that bootstrap pod is what's serving out the Ignition to the masters.  We don't have masters online until after `echo "etcd cluster up. Killing etcd certificate signer..."` (right?)

I believe we won't even have the `machineconfigs` CRD in the cluster until the MCO comes online.  So we will end up waiting in `openshift.service` for that - similar to how we wait for the cluster API objects and inject those.

Reply:
> > but I don't see any installer code obviously populating it.
>
> AIUI it's the static pod which we are just removing above this code:
`/etc/kubernetes/manifests/machineconfigoperator-bootstrap-pod.yaml`

So can we update that code to make these alterations?  It's maybe [here][1] calling [here][2]?

> So we will end up waiting in `openshift.service` for that - similar to how we wait for the cluster API objects and inject those.

I'm not sure how this comes into this pull request, but I'm pushing to get `openshift.service` and `openshift.sh` functionality moved into [`cluster-bootstrap`][3].  See #1147, which I'm going to reroll after `cluster-bootstrap` picks up openshift/library-go#220.

[1]: https://github.com/openshift/machine-config-operator/blob/c5f479e7ae15dfb579c2cf2b1575ceba9d0098e0/pkg/controller/bootstrap/bootstrap.go#L100-L133
[2]: https://github.com/openshift/machine-config-operator/blob/c5f479e7ae15dfb579c2cf2b1575ceba9d0098e0/pkg/controller/render/render_controller.go#L499-L548
[3]: https://github.com/openshift/cluster-bootstrap

Reply:
> So can we update that code to make these alterations?

Mmm...this code needs to generate content to inject into the target cluster.  The static pod is running on the bootstrap before the cluster, right?  

> but I'm pushing to get openshift.service and openshift.sh functionality moved into cluster-bootstrap.

That makes sense to me...I was surprised at the low-tech nature of `openshift.sh`.